ledger/accountdb.go

```diff
@@ -646,6 +777,25 @@ func accountsNewRound(tx *sql.Tx, updates map[basics.Address]accountDelta, creat
 			return
 		}
 
+		// Delete online accounts if they are no longer online.
+		// This takes advantage of the fact that a zero account
+		// value has Status=Offline, so this covers both deletion
+		// of accounts as well as switching to non-online state.
+		if data.old.Status == basics.Online && data.new.Status != basics.Online {
```

Comment:
this is a correct predicate, as it can safely skip the test for data.old.IsZero() because 0 == basics.Offline